doc/security/sensor-threat.rst

```diff
@@ -0,0 +1,327 @@
+
+3. **Root certificate list**. In order to authenticate the cloud service
+   (server), the IoT device must have a list of root certificates that
+   are allowed to sign the certificate on the server. For cloud-provider
+   based services, this list will generally be provided by the service
+   provider. Because the root certificates can expire, and possibly be
+   revoked, this list will need to be periodically be updated.
+
+4. **Client secrets**. To authenticate the client to the service, the
+   client must posess some kind of secret. This is general a private
+   key, usually either an RSA key or an EC private key. When
+   establishing communication with the server, the device will use
+   this secret either as part of the TLS establishment, or to sign a
+   message used in the communication.
+
+   This secret is generally generated by the service provider, or by
+   software running elsewhere, and must be securely installed on the
+   device. Policy may dictate that this secret be replaced
+   periodically, which will require a way to update the client secret.
+   Typically, the service will allow two or three active keys to allow
+   this update to proceed while the old key is used.
+
+5. **Current date/time**. TLS certificate verification requires
+   knowledge of the current date and time in order to determine if the
+   current time falls within the certificate’s current validity time.
+   Also, token based client authentication will generally require the
+   client to sign a message containing a time window that the token is
+   valid. Certificate validation requires the device notion of date and
+   time to be accurate within a day or so. Token generation generally
+   requires the time to be accurate within 5-10 minutes.
+
+6. **Sensor data**. The data received from the sensor itself, and
+   delivered to the service shall be delivered without modification or
+   tampering.
+
+7. **Device configuration**. Various configuration data, such as the
+   hostname of the service to connect to, the address of a time server,
+   frequency and parameters of when sensor data is sent to the service
+   and other need to be kept by the device. This configuration data will
+   need to be updated periodically as the configuration changes. Updates
+   should be allowed only from authorized parties.
+
+8. **Logs**. In order to assist with analysis of security issues, the
+   device shall log information about security-pertinent events. IoT
+   devices generally have limited storage, and as such, these logs need
+   to be carefully selected. It may also be possible to send these log
+   events to the cloud service where they can be stored in a more
+   resource-available environment. Types of events that should be logged
+   include:
+
+   a. **Firmware image updates**. The system should log the download of
+      new images, and when an image is successfully updated.
+
+   b. **Client secret changes**. And new client secrets should be
```

Comment:
the previous item had a complete sentence as the explanation, so may this should read:

    b. **Client secret changes**. Changes and new client secrets should be logged.